src/com/ableton/VirtualEnv.groovy

```diff
@@ -12,7 +12,10 @@ class VirtualEnv implements Serializable {
 
   VirtualEnv(script, String python) {
     this.script = script
+    assert script
```

Comment:
is this here for debugging, or is it meant for production code? I know for python, raw `asserts` are seen as less-good in prod code and better error reporting is usually desired. is that the same for groovy?

Reply:
also, can we move both `assert script` and `assert python` to the top of the function? it looks like they're checking the results of a computation, when they're really checking the inputs

Reply:
I'm not sure about Groovy code in general, but at least with respect to Jenkins pipelines I recall reading somewhere (though I can't find the link) that `assert` is a good practice since it's relatively cheap and will cause a build to fail for the right reasons.

In any case, for a shared library I think it's almost always appropriate to sanity check inputs since you have no idea how the code is being called.

Reply:
Also I agree that the `assert` statements should be together, I'll push a fixup for that.

Reply:
👍